src/pages/version/downstreamTasks/DownstreamProjectTable.tsx

```diff
@@ -0,0 +1,161 @@
+import { useRef, useMemo } from "react";
+import { LeafyGreenTable, useLeafyGreenTable } from "@leafygreen-ui/table";
+import {
+  ColumnFiltersState,
+  Filters,
+  Sorting,
+  SortingState,
+} from "@tanstack/react-table";
+import { useParams } from "react-router-dom";
+import { usePatchAnalytics, useVersionAnalytics } from "analytics";
+import { BaseTable } from "components/Table/BaseTable";
+import TableControl from "components/Table/TableControl";
+import { TablePlaceholder } from "components/Table/TablePlaceholder";
+import TableWrapper from "components/Table/TableWrapper";
+import { onChangeHandler } from "components/Table/utils";
+import { getColumnsTemplate } from "components/TasksTable/Columns";
+import { TaskTableInfo } from "components/TasksTable/types";
+import { SortDirection, TaskSortCategory } from "gql/generated/types";
+import { useTaskStatuses } from "hooks";
+import { Action } from "./reducer";
+
+const { getDefaultOptions: getDefaultFiltering } = Filters;
+const { getDefaultOptions: getDefaultSorting } = Sorting;
+
+interface DownstreamTasksTableProps {
+  childPatchId: string;
+  count: number;
+  dispatch: (action: Action) => void;
+  isPatch: boolean;
+  limit: number;
+  loading: boolean;
+  page: number;
+  taskCount: number;
+  tasks: TaskTableInfo[];
+}
+
+export const DownstreamTasksTable: React.FC<DownstreamTasksTableProps> = ({
+  childPatchId,
+  count,
+  dispatch,
+  isPatch,
+  limit,
+  loading,
+  page,
+  taskCount,
+  tasks,
+}) => {
+  const { id: versionId } = useParams<{ id: string }>();
+  const { sendEvent } = (isPatch ? usePatchAnalytics : useVersionAnalytics)(
+    versionId,
+  );
+
+  const { baseStatuses: baseStatusOptions, currentStatuses: statusOptions } =
+    useTaskStatuses({
+      versionId: childPatchId,
+    });
+
+  const onFilterChange = (filterState: ColumnFiltersState) => {
+    filterState.forEach(({ id, value }) => {
+      if (id === TaskSortCategory.Name) {
+        dispatch({ type: "setTaskName", task: value as string });
+      }
+      if (id === TaskSortCategory.Status) {
+        dispatch({ type: "setStatuses", statuses: value as string[] });
+      }
+      if (id === TaskSortCategory.BaseStatus) {
+        dispatch({ type: "setBaseStatuses", baseStatuses: value as string[] });
+      }
+      if (id === TaskSortCategory.Variant) {
```

Comment:
[nit] Should these use `else if` to avoid checking each statement on every loop? Or switch case

Reply:
updated!